.circleci/config.yml

```diff
@@ -0,0 +1,33 @@
+version: 2.1
+
+jobs:
+  build:
+    docker:
+      - image: nixos/nix:2.3
+    steps:
+      - checkout
+      - run:
+          name: Install build script
+          command: nix-env -f default.nix -iA build
```

Comment:
This requires that our nix expression in default.nix has an attribute `build`, which it currently cannot have I believe.

We have two options:
* Split the derivation at the end, which is an environment intended purely for nix-shell, into its own `shell.nix` file (see postgrest for this pattern). Then we fix default.nix to return a set at the end: `... in { inherit build spellcheck; }` (which is the same as `{ build = build; spellcheck = spellcheck; }`.
* We can be a bit sneaky and just set these attributes on the derivation we are already returning. We would notice at Nix eval time if that broke anything. To do that, we can just append `// { inherit build spellcheck; }` to default.nix ("merge the set before with this new set that has build and spellcheck as attributes, with the right hand side overriding the left hand side if needed").

Reply:
I think I did something inbetween. What do you think?

(Only reading your comment right now after pushing the latest changes)